File: 백준/Silver/11723. 집합/집합.java
import java.io.*;
import java.util.*;

public class Main {
	static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
	static int M,x;
	static Set<Integer> s = new HashSet<>();
	static StringTokenizer st;
	static StringBuilder sb = new StringBuilder();
	
	public static void main(String[] args) throws IOException {
		M = Integer.parseInt(br.readLine());
		for(int i=0; i<M; i++) {
			st = new StringTokenizer(br.readLine());
			String inst = st.nextToken();
			if(st.hasMoreTokens()) x = Integer.parseInt(st.nextToken());
			switch(inst) {
			case "add":
				s.add(x);
				break;
			case "remove":
				s.remove(x);
				break;
			case "check":
				sb.append(s.contains(x) ? 1 : 0).append('\n');
				break;
			case "toggle":
				if(s.contains(x)) s.remove(x);
				else s.add(x);
				break;
			case "all":
				for(int j=1; j<=20; j++) s.add(j);
				break;
			case "empty":
				s.clear();
				break;
			}
		}
		bw.write(sb.toString());
		bw.close();
		br.close();
	}
}

File: 백준/Silver/11723. 집합/집합.cc
#include <bits/stdc++.h>
using namespace std;
int M,x,mask;
string S;
int main() {
    ios_base::sync_with_stdio(0); cin.tie(0);
    cin >> M;
    for(int i=0; i<M; i++) {
        cin >> S;
        if(S=="add") {
            cin >> x;
            mask|=(1<<x);
        }
        else if(S=="remove") {
            cin >> x;
            mask&=~(1<<x);
        }
        else if(S=="check") {
            cin >> x;
            cout << (mask&(1<<x) ? 1 : 0) << '\n';
        }
        else if(S=="toggle") {
            cin >> x;
            mask^=(1<<x);
        }
        else if(S=="all") {
            mask=(1<<21)-1;
        }
        else {
            mask=0;
        }
    }
    return 0;
}